process/outportBlockConverter.go

```diff
@@ -0,0 +1,174 @@
+		return nil, fmt.Errorf("failed to cast developer fees: %w", err)
+	}
+
+	shardOutportBlock.BlockData.Header = &data.Header{
+		Nonce:              header.Header.Nonce,
+		PrevHash:           header.Header.PrevHash,
+		PrevRandSeed:       header.Header.PrevRandSeed,
+		RandSeed:           header.Header.RandSeed,
+		PubKeysBitmap:      header.Header.PubKeysBitmap,
+		ShardID:            header.Header.ShardID,
+		TimeStamp:          header.Header.TimeStamp,
+		Round:              header.Header.Round,
+		Epoch:              header.Header.Epoch,
+		BlockBodyType:      data.Type(header.Header.BlockBodyType),
+		Signature:          header.Header.Signature,
+		LeaderSignature:    header.Header.LeaderSignature,
+		MiniBlockHeaders:   miniBlockHeaders,
+		PeerChanges:        peerChanges,
+		RootHash:           header.Header.RootHash,
+		MetaBlockHashes:    header.Header.MetaBlockHashes,
+		TxCount:            header.Header.TxCount,
+		EpochStartMetaHash: header.Header.EpochStartMetaHash,
+		ReceiptsHash:       header.Header.ReceiptsHash,
+		ChainID:            header.Header.ChainID,
+		SoftwareVersion:    header.Header.SoftwareVersion,
+		AccumulatedFees:    accumulatedFees,
+		DeveloperFees:      developerFees,
+		Reserved:           header.Header.Reserved,
+	}
+
+	scheduledAccumulatedFees, err := o.castBigInt(header.ScheduledAccumulatedFees)
+	if err != nil {
+		return nil, fmt.Errorf("failed to cast scheduled accumulated fees: %w", err)
+	}
+	scheduledDeveloperFees, err := o.castBigInt(header.ScheduledDeveloperFees)
+	if err != nil {
+		return nil, fmt.Errorf("failed to cast scheduled developer fees: %w", err)
+	}
+
+	shardOutportBlock.BlockData.ScheduledRootHash = header.ScheduledRootHash
+	shardOutportBlock.BlockData.ScheduledAccumulatedFees = scheduledAccumulatedFees
+	shardOutportBlock.BlockData.ScheduledDeveloperFees = scheduledDeveloperFees
+	shardOutportBlock.BlockData.ScheduledGasProvided = header.ScheduledGasProvided
+	shardOutportBlock.BlockData.ScheduledGasPenalized = header.ScheduledGasPenalized
+	shardOutportBlock.BlockData.ScheduledGasRefunded = header.ScheduledGasRefunded
+
+	return shardOutportBlock, nil
+}
+
+// HandleMetaOutportBlock will convert an outport.OutportBlock to data.MetaOutportBlock.
+func (o *outportBlockConverter) HandleMetaOutportBlock(outportBlock *outport.OutportBlock) (*data.MetaOutportBlock, error) {
+	headerType := outportBlock.BlockData.HeaderType
+
+	// check if the header type is supported by this function.
+	if headerType != string(core.MetaHeader) {
+		return nil, fmt.Errorf("cannot convert to meta outport block. header type: %s not supported", outportBlock.BlockData.HeaderType)
+	}
+
+	// marshal with gogo, since the outportBlock is gogo protobuf (coming from the node).
+	bytes, err := o.gogoProtoMarshalizer.Marshal(outportBlock)
+	if err != nil {
+		return nil, fmt.Errorf("marshal metaBlockCaster error: %w", err)
+	}
+
+	// unmarshall into google protobuf. This is the proto that will be used in firehose.
```

Comment:
can you remove the reference to firehose in the comment? the converter will be used also for other RPC integrations

Reply:
changed.